stix_shifter_modules/trustar/stix_transmission/api_client.py

```diff
@@ -0,0 +1,376 @@
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from .error_mapper import ErrorMapper
+import requests
+import requests.auth
+import json
+import base64
+
+class APIClient():
+
+    def __init__(self, connection, configuration):
+        self.base_url = "https://{}".format(connection['host'])
+        self.headers = dict()
+        self.auth = configuration["auth"]
+        self.client = RestApiClient(connection['host'])
+        self.search_query = ""
+        self.headers["Content-Type"] = "application/json"
+        self.report_limit = configuration['report_limit']
+        self.max_number_pages = configuration['max_number_pages']
+        self.results_code = 200
+        self.success = True
+
+
+    def ping_data_source(self):
+        # Pings the data source
+        response = self.get_token()
+        return response
+
+    def get_search_results(self, search_id, range_start=None, range_end=None):
+        # Return the search results. Results must be in JSON format before being translated into STIX
+        response = {}
+        try:
+            response = self._get_results(search_id, range_start, range_end)
+        except Exception as e:
+            response = {'success':False, 'code': 2000, 'data': []}
```

Comment:
this response doesn't make sense to the user. specially code `'code': 2000`. I would suggest you to look at other existing modules for exception handlings.

Reply:
@mdazam1942
Implemented as suggested review it and merge it.